pkg/network/capability/capability.go

```diff
@@ -59,13 +70,14 @@ type Capability struct {
 func (c *Capability) DecodeBinary(br *io.BinReader) {
 	c.Type = Type(br.ReadB())
 	switch c.Type {
+	case ArchivalNode:
+		c.Data = &Archival{}
 	case FullNode:
 		c.Data = &Node{}
 	case TCPServer, WSServer:
 		c.Data = &Server{}
 	default:
-		br.Err = errors.New("unknown node capability type")
-		return
+		c.Data = &Unknown{}
```

Comment:
And now I think that we need to use ReservedFirst/ReservedLast to explicitly return an error if unexpected (and unreserved) type was accepted. Follow the https://github.com/nspcc-dev/neo-go/blob/64c4de4e38de24bc12741da018ded34ca6c05fd7/pkg/core/transaction/attribute.go#L67-L72

Reply:
That's exactly what we're trying to avoid to be able to roll out new capabilities faster. If this new capability follows the encoding rules old nodes could safely ignore it while new ones will do whatever they want.

Reply:
> If this new capability

The problem that current verification code doesn't distinguish reserved new capability from the garbadge. But if you think that it's sufficient for the current stage, then I'm OK with it.

Reply:
It's not sufficient for the current stage, it's a perfectly fine behavior forever. If we had it in 3.7, 3.8 could send Archival immediately and it'd be fine for 3.7 nodes.